AWSCore/Service/AWSService.m

```diff
@@ -478,6 +488,55 @@ + (NSString *)regionNameFromType:(AWSRegionType)regionType {
     }
 }
 
++ (AWSRegionType)regionTypeFromName:(NSString *)regionName {
```

Comment:
Check if you can reuse this - [aws_regionTypeValue](https://github.com/aws-amplify/aws-sdk-ios/blob/master/AWSCore/Utility/AWSCategory.m#L486)

Reply:
This is exactly what we need here. Thanks